src/main/java/roomescape/waiting/dto/request/WaitingRequest.java

```diff
@@ -0,0 +1,49 @@
+package roomescape.waiting.dto.request;
+
+import roomescape.exception.BadRequestException;
+import roomescape.exception.ExceptionMessage;
+import roomescape.theme.domain.Theme;
+import roomescape.time.domain.Time;
+import roomescape.waiting.domain.Waiting;
+
+import java.time.LocalDate;
+
+public record WaitingRequest(
+        LocalDate date,
+        long time,
+        long theme
+) {
+    public WaitingRequest {
+        validateDate(date);
+        validateTheme(theme);
+        validateTime(time);
+    }
+
+    private void validateDate(final LocalDate date) {
+        if (date == null) {
+            throw new BadRequestException(ExceptionMessage.INVALID_DATE.getMessage());
+        }
+    }
+
+    private void validateTheme(final Long themeId) {
+        if (themeId == null) {
+            throw new BadRequestException(ExceptionMessage.INVALID_THEME.getMessage());
+        }
+    }
+
+    private void validateTime(final Long timeId) {
+        if (timeId == null) {
+            throw new BadRequestException(ExceptionMessage.INVALID_TIME.getMessage());
+        }
+    }
+
+    public Waiting toWaiting(long memberId, String name, Time time, Theme theme) {
+        return new Waiting(
+                memberId,
+                name,
+                date,
+                time,
+                theme
+        );
```

Comment:
진짜 사소한 궁금증이긴 한데 정말 궁금해서,,,
아래와 같이 사용하지 않고 개행을 다 넣어준 이유가 있을까요?
```suggestion
        return new Waiting(memberId, name, date, time, theme);
```

Reply:
인자가 많아서 가독성을 위해 순간적으로 개행을 한 것 같은데, 매개변수들 이름이 다 짧아서 개행이 굳이긴 하네요,,허허허